zcash_client_backend/src/data_api/wallet.rs

```diff
@@ -955,27 +1058,28 @@ where
                     .output_index(i)
                     .expect("An output should exist in the transaction for each Sapling payment.");
 
-                let received_as = if let Recipient::InternalAccount(
-                    account,
-                    PoolType::Shielded(ShieldedProtocol::Sapling),
-                ) = recipient
-                {
-                    build_result
-                        .transaction()
-                        .sapling_bundle()
-                        .and_then(|bundle| {
-                            try_sapling_note_decryption(
-                                &internal_ivk,
-                                &bundle.shielded_outputs()[output_index],
-                                consensus::sapling_zip212_enforcement(params, min_target_height),
-                            )
-                            .map(|(note, _, _)| (account, note))
-                        })
-                } else {
-                    None
-                };
+                let recipient = recipient
+                    .map(|pool| {
```

Comment:
I was confused by this for at least five minutes, trying to figure out why `recipient` was optional and why this code was being significantly altered, until I realised that this is the new `Recipient::map` method that only affects `Recipient::InternalAccount`. If we're going to use this approach, please rename it for clarity (to e.g. `Recipient::map_internal_account`).

Reply:
It's just a functor! But fine.